Display appropriate message in all invoice action

kushal-dreamspring opened the pull request:
- Display 'No Invoice Found' if invoice not present
- Display correct invoice number and exit time if invoice is present
- Fix formatting of entry and exit time using strftime
- Add unit to duration of car parked

kushal-dreamspring committed:
Display appropriate message in all invoice action
- Display 'No Invoice Found' if invoice not present
- Display correct invoice number and exit time if invoice is present

jinifrancis commented on app.rb:
```diff
@@ -51,14 +51,18 @@ def parser_options
+    else
+      puts 'No Invoice Found'
+    end
   end
```
You can move the `print_invoice` method to the invoice controller instead of keeping it in `app.rb`. It is an action related to invoice. Or if it is a common method to be used by multiple controllers, create a helper module that can be included in both the controllers, so that the same method can be used by both the controllers.

kushal-dreamspring replied:
Should I also move print_all_invoices to invoice controller??

jinifrancis replied:
Yes. I see that you have already moved it in your new commit. Any action specific to a model, should ideally be in the controller specific to that model. If you have an invoice controller, any action specific to that should be routed to that invoice controller.

jinifrancis commented on app.rb:
```diff
@@ -51,14 +51,18 @@ def parser_options
+      puts "
 Invoice Details:
 Invoice number: #{invoice[:id]}
 Registration Number: #{invoice[:registration_number]}
 Entry Time: #{invoice[:entry_time]}
 Exit Time: #{invoice[:exit_time]}
 Duration: #{invoice[:duration]}
```
These 4 attributes are part of car - `invoice[registration_number]` looks incorrect. You can get the car from invoice and then use the car object to print these here.

kushal-dreamspring replied:
In my models, apart from the registration number, all the other fields are part of the invoice model. The reasoning being a car may enter the parking more than once.That is why I have done it this way

jinifrancis replied:
Got it. Further changes can be looked into when I do the code review. I'll check the functionality and proceed with this PR if it is working.

jinifrancis commented on controller/invoice.rb:
```diff
@@ -3,10 +3,10 @@
 class Controller
   def all_invoices
-    DATABASE[:invoices].join(:cars, id: :car_id).all
+    Car.join(:invoices, car_id: :id).all
```
Fetch the invoice and not the car from Invoice controller.

jinifrancis commented on controller/invoice.rb:
```diff
@@ -3,10 +3,10 @@
 
   def invoice(invoice_id)
-    DATABASE[:cars].join(DATABASE[:invoices].where(id: invoice_id), car_id: :id).first
+    Car.join(Invoice.where(id: invoice_id), car_id: :id).first
```
Fetch and return the invoice and not the car in the Invoice controller

jinifrancis reviewed (changes requested):
Here also I am not able to see related specs for the code updates made here. Can you add the necessary specs for these actions?

kushal-dreamspring committed:
Move invoice related actions to invoice controller

- Move print_invoice method to invoice controller
- Move print_all_invoices to invoice controller
- Add spec for no invoice found for print_invoice action

jinifrancis reviewed (changes requested):
Can you add the unit of time when you print the duration?
Ex: 10 seconds instead of saying 10

jinifrancis commented:
Also, make the entry time and exit time into a readable format when printing out details of the invoice.
Ex: July 3, 2023 17:37:38 IST instead of 2023-07-03 17:37:38 +0530

kushal-dreamspring committed:
Fix formatting of print_invoice and print_all invoice messages

- Fix formatting of entry and exit time using strftime
- Add unit to duration of car parked

kushal-dreamspring committed:
Add no invoice check to print_all_invoice action

jinifrancis commented:
All specs are not passing in this PR. Please check and fix the issue.
<img width="797" alt="Screenshot 2023-07-11 at 4 26 54 PM" src="https://github.com/kushal-dreamspring/parking_lot_rb/assets/17831956/e6f94616-47ae-4cc7-b1dd-49239b00f222">

kushal-dreamspring committed:
Merge branch main into fix/invoice_display_messages

jinifrancis reviewed (approved):
You can merge this PR!
